hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastJoiner.java

```diff
@@ -148,7 +141,7 @@ public void onMessage(Object msg) {
                 logger.warning(e);
             }
         } finally {
-            node.multicastService.removeMulticastListener(listener);
+            splitBrainJoinMessages.clear();
```

Comment:
I cannot think any problem with processing each message. There are already checks/validations for merge process. This is only discovery part.